src/js/plugins/trim.js

```diff
@@ -0,0 +1,229 @@
+// ==========================================================================
+// Plyr Trim control
+// ==========================================================================
+
+import { createElement, toggleClass, toggleHidden } from '../utils/elements';
+import { on, triggerEvent } from '../utils/events';
+import is from '../utils/is';
+
+class Trim {
+    constructor(player) {
+        // Keep reference to parent
+        this.player = player;
+        this.config = player.config.trim;
+        this.loaded = false;
+        this.trimming = false;
+        this.trimLength = 30;
+        this.tool = {
+            bar: null,
+            leftThumb: null,
+            rightThumb: null,
+            editing: null,
+        };
+
+        // Handle event (incase user presses escape etc)
+        on.call(this.player, document, () => {
+            this.onChange();
+        });
+
+        // Update the UI
+        this.update();
+
+        // Prevent the trim tool from being added until the player is in a playable state
+        // If the user has pressed the trim tool before this event has fired, show the tool
+        this.player.once('canplay', () => {
+            this.loaded = true;
+            if (this.trimming) {
+                this.createTrimTool();
+            }
+        });
+    }
+
+    // Determine if trim is enabled
+    get enabled() {
+        const { config } = this;
+        return config.enabled && this.player.isHTML5 && this.player.isVideo;
+    }
+
+    // Get active state
+    get active() {
+        if (!this.enabled) {
+            return false;
+        }
+
+        return this.trimming;
+    }
+
+    // Get trim range in seconds
+    getTrimRange() {
```

Comment:
It would probably be a good idea to have the trim range in ms aswell.
You could have `getTrimRangeSeconds` that's calling `getTrimRangeMS() / 1000`